### ISSUE TITLE
False positive: `xct_specific_matcher` when using `==` to compare types

### ISSUE THREAD

**Issue by sdc-78** [realm/SwiftLint#4990]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

I get a `xct_specific_matcher` warning when comparing two types using `XCTAssert( ... == ...)`, but types are not `Equatable` so `XCTAssertEqual(..., ...)` cannot be used.

```swift
func testTypes1() {
    XCTAssert(String.self == String.self) // --> SwiftLint warning: XCTest Specific Matcher Violation: Prefer the specific matcher 'XCTAssertEqual' instead (xct_specific_matcher)
}

func testTypes2() {
    XCTAssertEqual(String.self, String.self) // --> compiler error: Type 'String.Type' cannot conform to 'Equatable'
}
```

### Environment

* SwiftLint 0.52.0
* Relevant configuration file part:

```yml
opt_in_rules:
  - all
```

**Comment by SimplyDanny** [comment]
This is a known shortcoming of this rule that must be accepted, unfortunately. The reason is that normal SwiftLint rules only operate on the Swift AST without knowing anything about types.

However, I can imagine some heuristics to reduce the amount of false positives. In you example, if any of the operands is obviously a type (`Something.self`) the rule should stay quiet.
